### PR TITLE
Added GameMode parser

### LINKED ISSUE #182: Add GameModeParser

### PR DESCRIPTION
**Pull request by ikfir**
### ❓ Type of change

- [x] ✨ New feature (adds functionality)

### 📚 Description
Added GameMode Parser with lazy mapping

Closes #182 

### 🧪 How Has This Been Tested?

Unit testing

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

## Summary by CodeRabbit

- **New Features**
  - Introduced a GameMode parser with lazy mapping for enhanced command handling.
  - Added support for known Minecraft versions (1.8 to 1.21.1) with corresponding protocol numbers.

- **Bug Fixes**
  - Resolved issues in the spigot module related to player visibility and command reloading.
  - Addressed a bug with sender type inversion in spigot commands.

- **Enhancements**
  - Improved command processing with lazy mapping and dynamic parsers.
  - Expanded documentation on commands and region boxes.
  - Enhanced tab completion options and added case insensitivity for the map-based parser.

- **Tests**
  - Added comprehensive unit tests for GameModeCommand functionality and tab completion features.

### CODE REVIEW

File: commands/spigot-platform/src/test/java/net/apartium/cocoabeans/commands/spigot/parsers/GameModeCommand.java [hunk 1]
```diff
@@ -18,12 +19,10 @@ public void selfSetGamemode(Sender sender, GameMode gamemode) {
...
     @ExceptionHandle(value = InvalidParserResponse.InvalidParserException.class, priority = 1)
     public boolean invalidParser(Sender sender, InvalidParserResponse invalidParserResponse) {
-        if (invalidParserResponse instanceof AmbiguousMappedKeyResponse ambiguousMappedKeyResponse) {
-            sender.sendMessage("Did you mean one of the following [" + String.join(", ", ambiguousMappedKeyResponse.getKeys()) + "]?");
+        if (!(invalidParserResponse instanceof AmbiguousMappedKeyResponse ambiguousMappedKeyResponse))
             return false;
-        }
 
-        sender.sendMessage("oh no");
+        sender.sendMessage("Did you mean one of the following [" + String.join(", ", ambiguousMappedKeyResponse.getKeys()) + "]?");
```
[the remark this answers is not in the document]
Comment (ikfir) [reply commands/spigot-platform/src/test/java/net/apartium/cocoabeans/commands/spigot/parsers/GameModeCommand.java:25]: I like it that way no need for nesting

File: commands/spigot-platform/src/test/java/net/apartium/cocoabeans/commands/spigot/parsers/GameModeCommand.java [hunk 2]
```diff
@@ -18,12 +19,10 @@
...
```
[the remark this answers is not in the document]
Comment (ikfir) [reply commands/spigot-platform/src/test/java/net/apartium/cocoabeans/commands/spigot/parsers/GameModeCommand.java]: Just a test but fine

File: commands/spigot-platform/src/main/java/net/apartium/cocoabeans/commands/spigot/parsers/GameModeParser.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        super(keyword, GameMode.class, priority, true, true);
+
+        Map<String, GameMode> gameModeMap = new HashMap<>();
+
+        for (GameMode gameMode : GameMode.values()) {
+            gameModeMap.put(gameMode.name().toLowerCase(), gameMode);
+            gameModeMap.put(gameMode.getValue() + "", gameMode);
+        }
+
+        this.gameModeMap = Collections.unmodifiableMap(gameModeMap);
```
Comment (liorsl) [inline commands/spigot-platform/src/main/java/net/apartium/cocoabeans/commands/spigot/parsers/GameModeParser.java:34]: better use Map.copyOf here

**Review by liorsl — CHANGES_REQUESTED** [review]

File: commands/spigot-platform/src/main/java/net/apartium/cocoabeans/commands/spigot/parsers/GameModeParser.java (hunk 3 above)
Comment (ikfir) [reply commands/spigot-platform/src/main/java/net/apartium/cocoabeans/commands/spigot/parsers/GameModeParser.java:34]: Why?

**Review by liorsl — APPROVED** [review]

**Review by liorsl — APPROVED** [review]

**Pull request merged**